main.py

```diff
@@ -70,42 +71,77 @@
-    ############# Topic Modelling #############
-    number_of_articles = 1000
-    number_of_topics = 100
-    number_of_top_words = 10
+    ############# Topic Modeling #############
+    number_of_articles = 500
+    number_of_topics = 10
+    number_of_top_words = 5
     ds_passes = 2
+    match_score = 0.9
+
+    # Data
+    french_series = sentiment_df[sentiment_df['language'] == 'fr']
+    german_series = sentiment_df[sentiment_df['language'] == 'de']
+
+    # First rows
+    # german_series = german_series.iloc[:number_of_articles]['content']
+    # french_series = french_series.iloc[:number_of_articles]['content']
+
+    # Take random
+    german_series = german_series.sample(n=number_of_articles)['content']
+    french_series = french_series.sample(n=number_of_articles)['content']
+
+
+    ### LSA ###
+    logger.info("LSA")
 
     # French
-    french_series = sentiment_df[sentiment_df['language'] == 'fr'].iloc[:number_of_articles]['content']
-    #french_series = sentiment_df[sentiment_df['language'] == 'fr']['content']
-    french_model, _, _ = TopicModellingService.fit_lda(texts=french_series, language="french", num_topics=number_of_topics, dataset_passes=ds_passes)
-    french_top_words_per_topic = TopicModellingService.lda_top_words_per_topic(model=french_model, n_top_words=number_of_top_words)
-    #model_path = TopicModellingService.save_gensim_model(model=french_model, language="french")
-    #model1 = TopicModellingService.load_gensim_model(language="french")
-    #model2 = TopicModellingService.load_gensim_model(full_model_path=model_path)
+    french_lsa_model, _, _ = TopicModelingService.fit_model(texts=french_series, language="french", num_topics=number_of_topics, dataset_passes=ds_passes, technique='lsa')
+    df_lsa_topics_fr = TopicModelingService.get_top_words_per_topic(model=french_lsa_model, n_top_words=number_of_top_words, technique='lsa')
+    FileService.df_to_csv(df=df_lsa_topics_fr, file_name="lsa_topics_fr")
+
+    # German
+    german_lsa_model, _, _ = TopicModelingService.fit_model(texts=german_series, language="german", num_topics=number_of_topics, dataset_passes=ds_passes, technique='lsa')
+    df_lsa_topics_de = TopicModelingService.get_top_words_per_topic(model=german_lsa_model, n_top_words=number_of_top_words, technique='lsa')
+    FileService.df_to_csv(df=df_lsa_topics_de, file_name="lsa_topics_de")
+
+    # Matching topics Query: French / Corpus: German
+    best_lsa_matches_by_sentence_transformer_fr_de, _, _, _ = TopicMatcherService.match_by_sentence_transformer(df_lsa_topics_de['Word'], df_lsa_topics_fr['Word'], number_of_top=number_of_top_words, match_score=match_score, print_matches=True)
+
+    # Matching topics Query: German / Corpus: French
+    best_lsa_matches_by_sentence_transformer_de_fr, _, _, _ = TopicMatcherService.match_by_sentence_transformer(df_lsa_topics_fr['Word'], df_lsa_topics_de['Word'], number_of_top=number_of_top_words, match_score=match_score, print_matches=True, invert=True)
```

Comment:
what happens with words that are not matched? Do you have some stats about that or a plan?